File: introductory_problems/coin_piles/CoinPiles.py
def _can_pile(left: int, right: int) -> bool:
    if left > right:
        return _can_pile(right, left)
    diff = right - left
    if diff > left:
        return False
    return (left - diff) % 3 == 0
    


n = int(input())
for i in range(n):
    left, right = [int(x) for x in input().split()]
    if _can_pile(left, right):
        print("YES")
    else:
        print("NO")

File: introductory_problems/coin_piles/CoinPiles.java
// Working program using BufferedReader
import java.io.*;
import java.util.*;

public class CoinPiles {
    public static void main(String[] args) throws IOException {
        Scanner input = new Scanner(System.in);
        int n = Integer.valueOf(input.nextLine());
        Writer writer = new PrintWriter(System.out);
        for (int i = 0; i < n; i++) {
            String[] nums = input.nextLine().split(" ");
            boolean res = findValue(Integer.parseInt(nums[0]), Integer.parseInt(nums[1]));
            if (res)  {
                writer.write("YES\n");
            } else {
                writer.write("NO\n");
            }
        }
        input.close();
        writer.flush();
    }

    private static boolean findValue(int left, int right) {
        if (left > right) {
            return findValue(right, left);
        } 
        int diff = right - left;
        if (diff > left) return false;
        return (left - diff) % 3 == 0;
    }
}